contrib/mesos/pkg/executor/service/podsource/podsource.go

```diff
@@ -0,0 +1,201 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package podsource
+
+import (
+	"k8s.io/kubernetes/contrib/mesos/pkg/executor"
+	"k8s.io/kubernetes/contrib/mesos/pkg/podutil"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/cache"
+	kubetypes "k8s.io/kubernetes/pkg/kubelet/types"
+
+	log "github.com/golang/glog"
+)
+
+type (
+	filterType int
+
+	podName struct {
+		namespace, name string
+	}
+
+	// Filter is invoked for each snapshot of pod state that passes through this source
+	Filter interface {
+		// Before is invoked before any pods are evaluated
+		Before(podCount int)
+		// Accept returns true if this pod should be accepted by the source; a value
+		// of false results in the pod appearing to have been removed from apiserver.
+		// It is allowable for a Filter to override the value of `*api.Pod`.
+		Accept(**api.Pod) bool
```

Comment:
Suggestion: Instead of accepting `**Pod`, why not have a signature like `Accept(*api.Pod) (bool, *api.Pod)` which returns a potentially new Pod spec or the original reference.The caller then simply calls `pod, accepted = filter.Accept(pod)`.

Honestly it's a nit but leaves the decision whether to replace the original reference to the caller.